src/caffe/test/test_data/generate_sample_data.py

```diff
@@ -36,23 +36,25 @@
     f['label'] = label
     f['label2'] = label2
 
-with h5py.File(script_dir + '/sample_data_2_gzip.h5', 'w') as f:
+with h5py.File(script_dir + '/sample_data_uint8_gzip.h5', 'w') as f:
     f.create_dataset(
         'data', data=data + total_size,
         compression='gzip', compression_opts=1
     )
     f.create_dataset(
         'label', data=label,
-        compression='gzip', compression_opts=1
+        compression='gzip', compression_opts=1,
+        dtype='uint8',
     )
     f.create_dataset(
         'label2', data=label2,
-        compression='gzip', compression_opts=1
+        compression='gzip', compression_opts=1,
+        dtype='uint8',
     )
 
 with open(script_dir + '/sample_data_list.txt', 'w') as f:
-    f.write(script_dir + '/sample_data.h5\n')
-    f.write(script_dir + '/sample_data_2_gzip.h5\n')
+    f.write('src/caffe/test/test_data/sample_data.h5\n')
+    f.write('src/caffe/test/test_data/sample_uint8_gzip.h5\n')
```

Comment:
Ah...thanks for the explanation, I didn't realize that was an issue when I merged #2274.  In retrospect I think we should have just stuck with the existing behavior which required this script to be run from Caffe root, unless we were going to move to running this script to generate the data on-the-fly at test time (rather than tracking the test data files as we do now), which I think is probably a better way to do it (but might be rather involved w.r.t. the potential payoff...).  Anyway, I agree with you & retract my previous suggestion, and I think we should probably revert most or all of #2274 since we can't use the `script_dir`-relative paths everywhere.